### ISSUE TITLE
yolov11 how to improve large object performace?

### ISSUE THREAD

**Issue by sjtu-cz** [ultralytics/ultralytics#19661]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

<img width="466" alt="Image" src="https://github.com/user-attachments/assets/d64bf551-e496-472d-b436-769c2792ad0a" />

I use yolo-obb train the doclayout model, But the performance on large targets is relatively poor. How can this be improved?

**Comment by UltralyticsAssistant** [comment]
👋 Hello @sjtu-cz, thank you for your interest in Ultralytics 🚀! We recommend exploring the [Docs](https://docs.ultralytics.com/) for guidance, where you'll find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) examples and answers to common questions.

If this is a custom training ❓ Question, please provide more details, such as:
- Training settings (hyperparameters, epochs, batch size, etc.)
- Dataset information, including image examples and annotations
- Training logs and performance metrics

For 🐛 Bug Reports, please include a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug effectively.

Join the Ultralytics community where it suits you best:
- Real-time chat on [Discord](https://discord.com/invite/ultralytics) 🎧  
- In-depth discussions on [Discourse](https://community.ultralytics.com/)  
- Share and learn on our [Subreddit](https://reddit.com/r/Ultralytics)  

## Upgrade

Ensure you’re using the latest release of `ultralytics` with all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) installed in a **Python>=3.8** environment:

```bash
pip install -U ultralytics
```

## Environments

YOLO can be run in these verified environments (all dependencies preinstalled):
- **Notebooks** with free GPUs: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM: [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI: [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**: [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>  

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are passing. These tests verify YOLO functionality across macOS, Windows, and Ubuntu daily.

This is an automated response to guide you in the right direction. An Ultralytics engineer will review your issue and assist further soon. Thank you for your patience! 🙏

**Comment by Y-T-G** [comment]
You can try

https://github.com/ultralytics/ultralytics/issues/11634#issuecomment-2367471426

**Comment by sjtu-cz** [comment]
> You can try
> 
> [#11634 (comment)](https://github.com/ultralytics/ultralytics/issues/11634#issuecomment-2367471426)

I changed the reg_max from 16 to 84, but there still have problem to predict high and huge object. while seg model don't have the problem. 
Is there a flaw in the prob_iou loss of rotation object detection?

**Comment by Y-T-G** [comment]
Did you use my branch to train?

**Comment by sjtu-cz** [comment]
> Did you use my branch to train?

No, I just change the reg_max parameter

**Comment by sjtu-cz** [comment]
> Did you use my branch to train?

Are there any other important changes?

**Comment by Y-T-G** [comment]
Yes

**Comment by sjtu-cz** [comment]
> Yes

ok. I will try your branch, thanks a lot, by the way,  Could you briefly describe the changes?

**Comment by Y-T-G** [comment]
I changed the TAL assignment process

**Comment by sjtu-cz** [comment]
still not work, I use img size 1344, the reg_max print is 16, and the high box can not well detect.

<img width="641" alt="Image" src="https://github.com/user-attachments/assets/b5ccc9ac-c71e-406e-90c0-bbe8cf2b12c8" />

**Comment by Y-T-G** [comment]
You probably need P6 model.

https://www.reddit.com/r/Ultralytics/comments/1gbyc47/detecting_objects_that_are_extra_small_or_extra/

**Comment by sjtu-cz** [comment]
same data yolo-seg can well predict the high object, Is there a flaw in the prob_iou loss of rotation object detection?

**Comment by sjtu-cz** [comment]
您好，你的邮件我已经收到~

**Comment by glenn-jocher** [comment]
Thank you for your confirmation. If you have any additional technical questions about improving YOLO11 OBB performance for large objects, feel free to continue the discussion here in the GitHub issue. This way, solutions can benefit the wider community as well.

**Comment by sjtu-cz** [comment]
您好，你的邮件我已经收到~

**Comment by glenn-jocher** [comment]
I see you've sent the same message again. If you have any further technical questions about YOLO11 OBB performance, please feel free to ask here in the issue.
